stairway/src/main/java/bio/terra/stairway/impl/StairwayImpl.java

```diff
@@ -105,6 +102,10 @@ public StairwayImpl(StairwayBuilder builder) throws StairwayExecutionException {
             ? "stairway" + UUID.randomUUID().toString()
             : builder.getStairwayName();
 
+    this.executor =
+        Optional.ofNullable(builder.getExecutor())
+            .orElse(new DefaultThreadPoolTaskExecutor(builder.getMaxParallelFlights()));
```

Comment:
This is maybe minor but I think it's slightly better practice to do
```suggestion
            .orElseGet(() -> new DefaultThreadPoolTaskExecutor(builder.getMaxParallelFlights()));
```

By using a supplier, you don't actually create the `DefaultThreadPoolTaskExecutor` until you need it.  It's a constructor so maybe not a huge deal but I wanted to call it out since I've been bit by this before.

Reply:
I'm glad you raised this, I hadn't made that connection and agree that it should be changed.  This is getting into territory where the optional chaining is hindering readability for me, so I'll revisit in full.